src/sentry/static/sentry/app/views/organizationIntegrations/IntegrationDirectorySentryAppRow.tsx

```diff
@@ -22,42 +19,32 @@ type Props = {
   app: SentryApp;
   organization: Organization;
   install?: SentryAppInstallation;
-  isOnIntegrationPage: boolean;
   ['data-test-id']?: string;
 };
 
-export default class SentryApplicationRow extends React.PureComponent<Props> {
-  static propTypes = {
-    app: SentryTypes.SentryApplication,
-    organization: SentryTypes.Organization.isRequired,
-    install: PropTypes.object,
-    isOnIntegrationPage: PropTypes.bool,
-  };
-
+export default class IntegrationDirectorySentryAppRow extends React.PureComponent<Props> {
   get isInternal() {
     return this.props.app.status === 'internal';
   }
 
-  hideStatus() {
-    //no publishing for internal apps so hide the status on the developer settings page
-    return this.isInternal && !this.props.isOnIntegrationPage;
+  get isUnpublished() {
+    return this.props.app.status === 'unpublished';
+  }
+
+  get isPublished() {
+    return !(this.isInternal || this.isUnpublished);
```

Comment:
@NisanthanNanthakumar I think it would be better to have: `return this.props.app.status === 'published';`